From d202ab7b699a2c3c83670014b3f8870fb07c7a0f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:42:21 -0400
Subject: [alsa] pcm: Fix the silence data for DSD formats

Message-id: <140972823573-438-patchset-jkysela@redhat.com>
Patchwork-id: 91137
O-Subject: [RHEL 7.1 PATCH 437/442] ALSA: pcm: Fix the silence data for DSD formats
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Right now we set 0 as the silence data for DSD_U8 and DSD_U16 formats,
but this is actually wrong.  0 is rather the most negative value.
Alternatively, we may take the repeating 0x69 pattern like ffmpeg
deploys.

Reference: https://ffmpeg.org/pipermail/ffmpeg-cvslog/2014-April/076427.html
Suggested-by: Alexander E. Patrakov <patrakov@gmail.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>

(picked from kernel: 94a988a8ab91c0cdabd2431281ec09dc52d92674)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/core/pcm_misc.c b/sound/core/pcm_misc.c
index 4560ca0..2c6fd80 100644
--- a/sound/core/pcm_misc.c
+++ b/sound/core/pcm_misc.c
@@ -142,11 +142,11 @@ static struct pcm_format_data pcm_formats[(INT)SNDRV_PCM_FORMAT_LAST+1] = {
  },
  [SNDRV_PCM_FORMAT_DSD_U8] = {
   .width = 8, .phys = 8, .le = 1, .signd = 0,
-  .silence = {},
+  .silence = { 0x69 },
  },
  [SNDRV_PCM_FORMAT_DSD_U16_LE] = {
   .width = 16, .phys = 16, .le = 1, .signd = 0,
-  .silence = {},
+  .silence = { 0x69, 0x69 },
  },
  /* FIXME: the following three formats are not defined properly yet */
  [SNDRV_PCM_FORMAT_MPEG] = {
-- 
1.7.1